### PR TITLE
ensure request count read only once

### PR DESCRIPTION
**Pull request by kalantar**
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>
ensure requestcount metric is read only once even when specified with and without namespace in the same experiment
addresses issue https://github.com/iter8-tools/iter8/issues/849

### CODE REVIEW

**Comment by kalantar** [comment]
As part of this pull request, metrics specific tests were moved from `experiment_controller_test.go` to `metrics_test.go`. A new test was added -- it is the last one in the file.

File: controllers/metrics_test.go [hunk 1]
```diff
@@ -15,17 +15,255 @@ limitations under the License.
...
+	var jqe string = "expr"
+	var url string = "url"
 
-	a, b = namespaceName("world")
-	assert.Nil(t, a)
-	assert.Equal(t, "world", b)
-}
+	var testName string
+	var testNamespace, metricsNamespace string
+	var goodObjective, goodObjective2, badObjective, reward *v2alpha2.Metric
```
Comment (sriumcp) [inline controllers/metrics_test.go:33]: Please rename goodObjective, and goodObjective2 as `goodMetric` and `goodMetric2` respectively.

**Review by sriumcp — CHANGES_REQUESTED** [review]
Minor change requested.

File: controllers/metrics_test.go (hunk 1 above)
Comment (sriumcp) [reply controllers/metrics_test.go:33]: Other than this minor change, this PR looks good to merge.

Comment (kalantar) [reply controllers/metrics_test.go]: renamed these and all the others as well.

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**